src/bodywork/workflow_execution.py

```diff
@@ -303,14 +303,15 @@ def _run_batch_stages(
     try:
         timeout = max(stage.max_completion_time for stage in batch_stages)
         k8s.monitor_jobs_to_completion(job_objects, timeout + TIMEOUT_GRACE_SECONDS)
+        _print_logs_to_stdout(namespace, job_name)
     except TimeoutError as e:
         _log.error("Some (or all) k8s jobs failed to complete successfully")
+        _print_logs_to_stdout(namespace, job_name)
         raise e
     finally:
         for job_object in job_objects:
             job_name = job_object.metadata.name
             _log.info(f"Completed k8s job for stage = {job_name}")
-            _print_logs_to_stdout(namespace, job_name)
```

Comment:
I don't understand this change. When `_print_logs_to_stdout(namespace, job_name)` is in the `finally` block then all logs should be printed out for each job when they've completed or timed out. 

Now the only logs that will be printed are for the last job in the loop defined on line 300. Is this correct or have I missed something?

Reply:
If there are no resources available to run a pod, then it won't generate any logs. So, I don't want to print empty pod logs to stdout. Thus, the logs will only now get printed if the job/deployment complete successfully or times-out.

Reply:
But you're right, I am missing a for loop...

Reply:
Cool, that makes sense. `k8s.monitor_jobs_to_completion` can also throw up a `BodyworkJobFailure` exception so I think we need to catch that exception and print the logs out for it as well (e.g. in the same block as TimeoutError).